plugins/node/opentelemetry-instrumentation-koa/src/utils.ts

```diff
@@ -56,10 +56,49 @@ export const getMiddlewareMetadata = (
  */
 export const isLayerIgnored = (
   type: KoaLayerType,
-  config?: KoaInstrumentationConfig
+  config?: KoaInstrumentationConfig,
+  name?: string
 ): boolean => {
-  return !!(
+  if (
     Array.isArray(config?.ignoreLayersType) &&
     config?.ignoreLayersType?.includes(type)
-  );
+  ) {
+    return true;
+  }
```

Comment:
Given that instrumentation.ts calls `isLayerIgnored` in two different ways -- once to check the `type` and once to check the `name` --  I think it would be cleaner to have two separate `isLayerTypeIgnored` and `isLayerNameIgnored` (or whatever names).

For example, I think the current `isLayerIgnored` has an issue: when it is called without a name: `isLayerIgnored(type, config)`, the code below that calls `satisfiesPattern` is still called, even though `name` is undefined. It shouldn't be called. Separating to two functions would make this cleaner.